File: telegram-bot-service/src/main/java/telegram/polling/HandlerRegistry.java
package telegram.polling;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class HandlerRegistry {

    /**
     * Command mapping
     */
    private final Map<String, UpdateHandler> commands = new HashMap<>();

    /**
     * Handler chain
     */
    private final List<UpdateHandler> handlers = new ArrayList<>();

    public HandlerRegistry register(UpdateHandler handler, String... aliases) {

        for (String alias : aliases) {
            commands.put(alias, handler);
        }
        return this;
    }

    public HandlerRegistry register(UpdateHandler handler) {
        handlers.add(handler);
        return this;
    }

    public UpdateHandler get(String alias) {
        return commands.get(alias);
    }

    public List<UpdateHandler> getHandlerChain() {
        return handlers;
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/bot/consumer/HandlerRegistry.kt
package com.github.bumblebee.bot.consumer

import java.util.*

class HandlerRegistry {

    /**
     * Command mapping
     */
    private val commands = HashMap<String, UpdateHandler>()

    /**
     * Handler chain
     */
    private val handlers = ArrayList<UpdateHandler>()

    val handlerChain: List<UpdateHandler>
        get() = handlers

    fun register(handler: UpdateHandler, aliases: List<String>): HandlerRegistry {

        aliases.forEach { alias -> commands.put(alias, handler) }
        return this
    }

    fun register(handler: UpdateHandler): HandlerRegistry {
        handlers.add(handler)
        return this
    }

    operator fun get(alias: String?): UpdateHandler? {
        return commands[alias]
    }
}
